### PR TITLE
tinc: fetch ed25519 public keys from uci

### PR DESCRIPTION
**Pull request by vgaetera**
Maintainer: @ErwanMAS
Run tested: OpenWrt 23.05.0 x86/64

Description:
Fetch Ed25519 public keys from UCI host sections.
Update options and syntax to current version.

### CODE REVIEW

**Comment by BKPepe** [comment]
This requires to bump ``PKG_RELEASE`` in the ``Makefile``.

**Comment by ErwanMAS** [comment]
`Name` is not redundant .  The goal is here to override the section name from uci-config .

This useful when you have the same hostname but for different network

**Comment by vgaetera** [comment]
OK, I corrected the code, but the Name option is [server-specific](https://tinc-vpn.org/git/browse?p=tinc;a=blob;f=src/tincctl.c;h=97b08cb55abe7f5815b25d72aaedacadfc08b096;hb=3217d5efb432f5a03beebd5d00b36392ec4b22ef#l1759), so there's no need to add it  inside host files.

File: net/tinc/files/tinc.init [hunk 1]
```diff
@@ -91,7 +91,7 @@ prepare_host() {
 	HOST_CONF_FILE="$TMP_TINC/$n/hosts/$s"
 	MANDATORY_PARAM_IN_UCI=0
 	[ ! -f "/etc/tinc/$n/hosts/$s" ] && {
-	        config_get pk "$s" "PublicKey"
+		config_get pk "$s" "Ed25519PublicKey"
 		config_get na "$s" "Name"
 		if [ -n "$pk" -a -n "$na" ] ; then
 		        HOST_CONF_FILE="$TMP_TINC/$n/hosts/$na"
```
Comment (ErwanMAS) [inline net/tinc/files/tinc.init:97]: i will propose this code : 
```
          config_get   pk_i "$s" "PublicKey"
          config_get   pk_f "$s" "PublicKeyFile"
          config_get   pked_i "$s" "Ed25519PublicKey"
          config_get   pked_f "$s" "Ed25519PublicKeyFile"
          config_get   na "$s" "Name"
          if [ -n "$na" ] ; then
                 HOST_CONF_FILE="$TMP_TINC/$n/hosts/$na"
          fi
          if [ -n "$pk_i" -o -n "$pk_f" -o -n "$pked_i" -o -n "pked_f" ] ; then
                 MANDATORY_PARAM_IN_UCI=1
          fi
          
```

Comment (stangri) [reply net/tinc/files/tinc.init]: More compact this way:
```
          if [ -n "${pk_i}${pk_f}${pked_i}{pked_f}" ] ; then
                 MANDATORY_PARAM_IN_UCI=1
          fi
```

**Review by ErwanMAS — APPROVED** [review]
i am ok with all changes in this PR

**Comment by ErwanMAS** [comment]
@stangri @BKPepe can you merge this request please ?

**Pull request merged**